src/test/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfigurationIntegrationTest.java

```diff
@@ -0,0 +1,66 @@
+package io.jenkins.plugins.opentelemetry;
+
+import io.jenkins.plugins.opentelemetry.semconv.JenkinsMetrics;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricExporterProvider;
+import io.opentelemetry.sdk.testing.exporter.InMemoryMetricExporterUtils;
+import io.opentelemetry.semconv.ServiceAttributes;
+import java.io.File;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.Optional;
+import jenkins.model.Jenkins;
+import org.junit.After;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.jvnet.hudson.test.Issue;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
+import static org.awaitility.Awaitility.await;
+import static org.hamcrest.Matchers.is;
+
+@WithJenkins
+public class JenkinsOpenTelemetryPluginConfigurationIntegrationTest {
```

Comment:
This is a new standalone class, because I didn't see any existing class that was better for the test. We could extend `BaseIntegrationTest`, but that class directly configures `JenkinsControllerOpenTelemetry` instead of going through `JenkinsOpenTelemetryPluginConfiguration`, so although we could use it, it would be doing some stuff we don't want, and the `JenkinsRule` field there is static, which makes it problematic for tests that want to change the global configuration. The `ConfigurationAsCode*Test` classes do check `JenkinsOpenTelemetryPluginConfigurationIntegration`, but they use CasC for the actual configuration, and we need to change the config in the test. Most of the other integration tests configure OTel via CasC and never change the configuration again.